pex/resolve/locker_patches.py

```diff
@@ -197,12 +197,12 @@ def patch_wheel_model():
 
     def get_abi_info(self):
         if hasattr(self, "abis"):
-            abis = list(self.abis)
+            abis = set(self.abis)
         else:
-            abis = [file_tag.abi for file_tag in self.file_tags]
+            abis = {file_tag.abi for file_tag in self.file_tags}
 
-        is_abi3 = ["abi3"] == abis
-        is_abi_none = ["none"] == abis
+        is_abi3 = {"abi3"} == abis
```

Comment:
N.B.: For a wheel like `psutil-5.9.8-cp36-abi3-manylinux_2_12_i686.manylinux2010_i686.manylinux_2_17_i686.manylinux2014_i686.whl`, there are 4 expanded tags:
+ cp36-abi3-manylinux_2_12_i686
+ cp36-abi3-manylinux2010_i686
+ cp36-abi3-manylinux_2_17_i686
+ cp36-abi3-manylinux2014_i686

That means the list on the LHS was `["abi3", "abi3", "abi3", "abi3"]` which did not match `["abi3"]`.

Reply:
As far as the `==` test vs a containment test - IIUC the wheel tag syntax is over-general here. It makes sense to have a dotted pyver tag; e.g. `cowsay-6.0-py2.py3-none-any.whl` and it makes sense to have a dotted platform tag (the case above is an example of this), but it never makes sense to have a dotted abi tag. That would mean a wheel is both, say, abi3 and cp312, but, AFAICT you can't have a native python binding that is both using the limited API and the targeted one at the same time.